src/main/java/org/jilt/internal/AbstractBuilderGenerator.java

```diff
@@ -333,13 +335,34 @@ protected final TypeVariableName mangleTypeParameter(TypeVariableName typeVariab
     private ParameterSpec setterParameterSpec(VariableElement attribute, TypeName parameterType) {
         ParameterSpec.Builder ret = ParameterSpec.builder(parameterType,
                 this.attributeSimpleName(attribute));
-        AnnotationMirror nullableAnnotation = this.firstAnnotationCalledNullable(attribute);
-        if (nullableAnnotation != null) {
-            ret.addAnnotation(AnnotationSpec.get(nullableAnnotation));
-        }
+        this.addAnnotationsToParam(ret, attribute.getAnnotationMirrors());
+        this.addAnnotationsToParam(ret, attribute.asType().getAnnotationMirrors());
         return ret.build();
     }
 
+    private void addAnnotationsToParam(ParameterSpec.Builder ret, List<? extends AnnotationMirror> annotationMirrors) {
```

Comment:
Let's just call the first parameter `param`, and the second parameter `annotations`:

```suggestion
    private void addAnnotationsToParam(ParameterSpec.Builder param, List<? extends AnnotationMirror> annotations) {
```